src/pugixml.cpp

```diff
@@ -5441,6 +5441,13 @@ namespace pugi
 		return impl::strcpy_insitu(_attr->name, _attr->header, impl::xml_memory_page_name_allocated_mask, rhs, size);
 	}
 
+#ifdef PUGI_HAS_STRING_VIEW
+	PUGI_IMPL_FN bool xml_attribute::set_name(const string_view_t rhs)
```

Comment:
Here and elsewhere `const` should not be used for arguments as they are passed by value

Reply:
This is my usual style (as much `const` as possible), although I would use east const.
I don't have any problems with changing this, but out of curiosity, could you elaborate on your rationale?
The fact that they are passed by value is orthogonal to the fact that the value is not changed.
And because it is not changed it can be made `const` and IMHO consequently should be made `const`.

Reply:
I personally do not subscribe to the "what could be made const should be" when that doesn't affect the interface; in this case, the caller behavior does not change based on whether the argument is const or not. I am aware of the argument that it can help prevent mistakes but I have not seen these happen much in practice so I view this as syntactic noise. You're welcome to disagree of course, but for consistency this `const` should be removed: pugixml doesn't have a single function with a const parameter that is passed by value, and the only uses of const locals are that where the argument is a *compile-time* const (when this code was written this seemed to  enhance readability, whether this is true or not is subjective).